core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -11958,9 +11958,9 @@ public final String getUserClaimValueWithID(String userID, String claim, String
 
         if (requireRolesAndGroups) {
             rolesAndGroups = getRoleListOfUserWithID(userID);
-        } else if (requireRoles || requireIntRole) {
+        } else if (requireIntRole) {
             rolesAndGroups = doGetInternalRoleListOfUserWithID(userID, "*");
-        } else if (requireGroups || requireExtRole) {
```

Comment:
Clarification:

Both `requireRoles` and `requireGroups` are boolean flags and indicate that either roles, groups, or both should be populated when sending user claims in the end.

In the same manner(before this PR is made), all the groups and roles were sent outside as the value of `role` claim. There is legacy logic in the kernel to separate out groups and roles though(identified as internal and external roles previously). `requireIntRole ` and `requireExtRole ` are flags which were used for that purpose. 

However, combining these legacy boolean flags and newer boolean flags can cause unwanted behaviour. For example if we dont remove flags like in here, then if getUserClaimValues method requests wso2.groups claim, not only wso2.groups claim but wso2.role claim also get populated because of the `OR` condition at line number 11963. Therefore these flags are removed in this manner, and if you inspect the code logic afterwards you may see groups and roles are correctly handled separately.